Islandora/islandora_scholar
# embargo roles whitelist

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1602
- Other Relevant Links: https://github.com/Islandora/islandora_scholar/pull/220 (the original pull request for this issue)
# What does this Pull Request do?

Allows a roles whitelist to be configured and applied to embargoes.
# What's new?

Three things:
- A configurable whitelist has been added to establish trusted roles with the 'can embargo any' permission that are added to embargoes when they are set via `islandora_scholar_embargo_set_embargo()`.
- A batch has been added to reapply existing embargoes. It can be triggered via the configurable whitelist form, if desired.
- A drush command has been added, `islandora_scholar_embargo_reapply_embargoes`, which also triggers the batch re-application.
# How should this be tested?
- The whitelist form should be tested to make sure the variables it sets take the form of an associative array pairing role IDs with role names.
- The batch should be tested to ensure embargoes on objects and datastreams are applied properly
- The drush command should also be tested to ensure it functions as expected.

As this has XACML security implications, we should be ensuring that when an embargo is created, a POLICY is applied that _only_ contains the roles in the whitelist. POLICYs for entire objects should have viewing _and_ management permissions for whitelisted roles.
# Additional Notes:

Care has been taken to structure the descriptions of things so that there's no ambiguity, since ambiguous and misleading documentation seems to be part of the source of this issue. Documentation within the module should probably be checked to make sure it's internally consistent and communicates clearly.

One of the things that is documented in this pull but bears repeating is that due to the way the Fedora-Drupal filter works, matching role IDs between sites that share the same Fedora represent a method of seeing objects that would otherwise be hidden. Whitelist roles should likely be unique and/or namespaced to mitigate this issue.
# Interested parties

From the original PR: @bryjbrown @rosiel @jordandukart
